internal/webhook/external/gitea_test.go

```diff
@@ -0,0 +1,179 @@
+package external
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/akuity/kargo/internal/indexer"
+)
+
+const giteaSigningKey = "mysupersecrettoken"
+
+func TestGiteaHandler(t *testing.T) {
+	const testURL = "https://webhooks.kargo.example.com/nonsense"
+
+	const testProjectName = "fake-project"
+
+	var validPushEvent = struct {
+		Repo struct {
+			Name string `json:"html_url"`
+		} `json:"repository"`
+	}{Repo: struct {
+		Name string `json:"html_url"`
+	}{Name: "https://gitea.com/example/repo"}}
```

Comment:
Nesting anonymous struct types can be clunky. (Notice you had to repeat `json:"html_url"` twice to make this work?) For the small chunk of JSON you want to get by marshaling this, I would just define it as a string literal instead.

```suggestion
const giteaWebhookRequestBodyImage = `
{
	"repository": {
		"clone_url": "https://gitea.com/example/repo.git"
	}
}`
```